versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractBinary.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.tests;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.protobuf.ByteString;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.versioned.persist.adapter.Compression;
+import org.projectnessie.versioned.persist.adapter.ContentAttachment;
+import org.projectnessie.versioned.persist.adapter.ContentAttachmentKey;
+import org.projectnessie.versioned.persist.adapter.ContentId;
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapter;
+
+/**
+ * Verifies that a big-ish number of keys, split across multiple commits works and the correct
+ * results are returned for the commit-log, keys, global-states. This test is especially useful to
+ * verify that the embedded and nested key-lists (think: full-key-lists in a commit-log-entry) work
+ * correctly.
+ */
+public abstract class AbstractBinary {
```

Comment:
How do we separate units from integration tests?

Reply:
For the database adapters it's basically a "unit test", if the database adapter is in memory (so H2 + the inmem adapter) - if it's a separate process (e.g. Mongo, Dynamo, Postgres), then it's an integration test. Similar for the tests via Quarkus.

Reply:
IT = maven-failsafe-plugin
Test* or *Test = maven-surefire-plugin

You mean this?